### ISSUE TITLE
feat(doc): add a learn-as-you-type experience on the documentation

### ISSUE THREAD

**Issue by s-pace** [sdispater/orator#365]
👋 team,

TL;DR: **Add same search experience as https://laravel.com/docs/7.x/**

I'm working at Algolia on the a project called [**DocSearch**](https://docsearch.algolia.com/) which goal is to enhance documentation websites with exhaustive, fast and relevant search. You might have seen DocSearch live already on websites like [Bootstrap](https://getbootstrap.com/), [Brew](https://brew.sh/) or [jQuery](https://jquery.com/).

We **use orator and we love it**, we want to give back to this great project.

[![demo of DocSearch + orator](https://s5.gifyu.com/images/Screen-Recording-2020-03-26-at-08.13-AM.gif)](https://docsearch.algolia.com/demo/?indexName=orator)

The way DocSearch works is by crawling your content, pushing the results into an Algolia index, and then requesting this index directly from the website front-end through JavaScript.

We'll take care of crawling your website and populating the Algolia index with the latest changes every 24h for you. You don't need to change anything to your deployment process. [All instructions to integrate it are available here](https://docsearch.algolia.com/demo/?indexName=orator).  Spoiler Alert: The only thing you need to **add are the following CSS and JS snippets that will bind the dropdown to your searchbox.**

We built DocSearch with the idea of giving back to the Open-Source community. This is why your [crawling configuration](https://github.com/algolia/docsearch-configs/blob/master/configs/orator.json) is available on GitHub if you want to change it. We also have our own [documentation](https://docsearch.algolia.com/) to help you tweak the dropdown to your needs. You'll also have access to *analytics on the most searched terms* or those with *no results*. All of this is of course **entirely free**.
